src/core/meshcop/dataset_manager_ftd.cpp

```diff
@@ -121,10 +121,17 @@ Error DatasetManager::HandleSet(Coap::Message &aMessage, const Ip6::MessageInfo
         VerifyOrExit(pendingTimestamp.IsValid());
     }
 
-    // verify the request includes a timestamp that is ahead of the locally stored value
-    timestamp = (type == Tlv::kActiveTimestamp) ? static_cast<Timestamp *>(&activeTimestamp)
-                                                : static_cast<Timestamp *>(&pendingTimestamp);
+    if (type == Tlv::kActiveTimestamp)
+    {
+        timestamp = static_cast<Timestamp *>(&activeTimestamp);
+    }
+    else
+    {
+        VerifyOrExit(pendingTimestamp.GetLength() > 0);
+        timestamp = static_cast<Timestamp *>(&pendingTimestamp);
+    }
```

Comment:
Quick note on this:
- At the top of this method, we do `pendingTimestamp.SetLength(0);` but this only sets the length and the TLV and it embedded `Timestamp` value would not be initialized.
- In the code right above this, we try to find and read the `pendingTimestamp` TLV from the message.
```c++
    if (Tlv::FindTlv(aMessage, pendingTimestamp) == kErrorNone)
    {
        VerifyOrExit(pendingTimestamp.IsValid());
    }
```
- But if the TLV  is not present in the message then without the change in this PR we can continue and  cast it to `Timestamp *` and use its value.